### PR TITLE
Notification status

### PR DESCRIPTION
**Pull request by tobygabriella**
- Implemented feedback like making enums for notification type
- Users can now see how many unread notifications they have on their notification center icon
- Unread notifications are on the top of list and have a distinguishing look
- ScreenRecording: https://www.loom.com/share/137f45e98b3f491cade2e82dc176eca8?sid=f78c02d2-ba74-4a36-a396-bb704cc2d346

### CODE REVIEW

File: Backend/index.js [hunk 1]
```diff
@@ -523,6 +531,41 @@ app.get("/notifications", verifyToken, async (req, res) => {
   }
 });
 
+app.post("/notifications/mark-as-read", verifyToken, async (req, res) => {
```
Comment (conniewu3) [inline Backend/index.js:534]: i would pull this out into a separate commit next time! so one for addressing previous comments (adding new enum, removing console.logs) and then another one for mark as read

File: Frontend/src/Components/Notifications/Notifications.jsx [hunk 2]
```diff
@@ -1,67 +1,52 @@
...
+    }
+  }, [notifications]);
 
-    fetchNotifications();
-
-    socket.on("notification", (notification) => {
-      setNotifications((prev) => [notification, ...prev]);
-    });
+  useEffect(() => {
+    const timer = setTimeout(() => {
```
Comment (conniewu3) [inline Frontend/src/Components/Notifications/Notifications.jsx:21]: maybe something we can think about doing is only marking as read when a notif is clicked on, but clearing the notif count when the notif page is opened. kind of like on workplace!

File: Frontend/src/Components/Notifications/Notifications.jsx [hunk 3]
```diff
@@ -1,67 +1,52 @@
...
 
 const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
 
 const Notifications = () => {
-  const { socket } = useSocket();
-  const [notifications, setNotifications] = useState([]);
-  const [isLoading, setLoading] = useState(true);
-  const [error, setError] = useState(null);
+  const { notifications, markAsRead } = useSocket();
+  const { isLoading, stopLoading } = useLoading();
```
Comment (conniewu3) [inline Frontend/src/Components/Notifications/Notifications.jsx:12]: should you startloading somewhere?

File: Backend/index.js (hunk 1 above)
Comment (tobygabriella) [reply Backend/index.js:534]: alrightt

File: Frontend/src/Components/Notifications/Notifications.jsx (hunk 2 above)
Comment (tobygabriella) [reply Frontend/src/Components/Notifications/Notifications.jsx:21]: That sounds like a good idea! would definitely implement

File: Frontend/src/Components/Notifications/Notifications.jsx (hunk 3 above)
Comment (tobygabriella) [reply Frontend/src/Components/Notifications/Notifications.jsx:12]: you're right!

File: Frontend/src/Components/Headers/AppHeader.jsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
-import React,{ useState } from 'react';
+import React,{ useState} from 'react';
 import { Link, useLocation, useNavigate } from "react-router-dom";
 import { useAuth } from '../Contexts/AuthContext';
 import './AppHeader.css';
 import { getInitials } from "../utils/initialsUtils";
+import { useSocket } from "../Contexts/SocketContext";
+
+const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
```
Comment (conniewu3) [inline Frontend/src/Components/Headers/AppHeader.jsx:8]: do you use this anywhere?

**Comment by conniewu3** [comment]
few small things but looks good!

File: Frontend/src/Components/Headers/AppHeader.jsx (hunk 4 above)
Comment (tobygabriella) [reply Frontend/src/Components/Headers/AppHeader.jsx:8]: oh no, just made the changes
